### PR TITLE
Add docs for jenkins custom cicd template

### PR DESCRIPTION
**Pull request by pohek321**
Did this for a customer awhile back, forgot to add it here in the official docs.

### CODE REVIEW

File: astro/ci-cd-templates/jenkins.md [hunk 1]
```diff
@@ -145,6 +145,57 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+
+    - `ASTRONOMER_KEY_ID`: Your Deployment API key ID
+    - `ASTRONOMER_KEY_SECRET`: Your Deployment API key secret
+    - `ASTRONOMER_DEPLOYMENT_ID`: The Deployment ID of your production deployment
+
+    Be sure to set the values for your API credentials as secret.
+
+2. At the root of your Astro Git repository, add a [Jenkinsfile](https://www.jenkins.io/doc/book/pipeline/jenkinsfile/) that includes the following script:
+
+    <pre><code parentName="pre">{`pipeline {
```
Comment (lzdanski) [inline astro/ci-cd-templates/jenkins.md:167]:
```suggestion
```text
{`pipeline {
```

File: astro/ci-cd-templates/jenkins.md [hunk 2]
```diff
@@ -145,6 +145,57 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+                    '''
+                }
+            }
+        }
+        post {
+            always {
+                cleanWs()
+            }
+        }
+    }`}</code></pre>
```
Comment (lzdanski) [inline astro/ci-cd-templates/jenkins.md:193]:
```suggestion
    }`}
    ```
```

**Review by lzdanski — CHANGES_REQUESTED** [review]
These changes look good! I saw one place where we might want to add a sentence about where or how users can find their Deployment information and there's a markdown formatting syntax change for the codeblock that's needed.

File: astro/ci-cd-templates/jenkins.md [hunk 3]
```diff
@@ -145,6 +145,57 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+
+- An Astro project that requires additional build-time arguments to build the Runtime image.
+
+1. In your Jenkins pipeline configuration, add the following parameters:
+
+    - `ASTRONOMER_KEY_ID`: Your Deployment API key ID
+    - `ASTRONOMER_KEY_SECRET`: Your Deployment API key secret
+    - `ASTRONOMER_DEPLOYMENT_ID`: The Deployment ID of your production deployment
+
+    Be sure to set the values for your API credentials as secret.
```
[the remark this answers is not in the document]
Comment (lzdanski) [reply astro/ci-cd-templates/jenkins.md:163]: Would the single branch and custom use the same method to find the parameters? Is it just the cli - [`astro deployment inspect`](https://docs.astronomer.io/astro/cli/astro-deployment-inspect) or go to the deployment in the Cloud UI?

File: astro/ci-cd-templates/jenkins.md (hunk 1 above)
Comment (pohek321) [reply astro/ci-cd-templates/jenkins.md]: These changes broke the CI/CD checks -- not sure what that `<pre>` tag does @lzdanski, maybe you could elaborate?

Comment (lzdanski) [reply astro/ci-cd-templates/jenkins.md]: Absolutely! So with markdown syntax, it accepts both markdown and HTML markup. The declaration `<pre><code>` is a way to declare to the framework that we're using HTML for that section, not markdown. However, the build errors are being reported from line 241 in that file, which isn't from the changes you made. It looks like it's happy with our syntax, but not happy going back to using HTML for codeblocks. 

Can I tinker with it and push a change back to your branch?
Screenshot 2023-05-23 at 4 20 43 PM

File: astro/ci-cd-templates/jenkins.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pohek321) [reply astro/ci-cd-templates/jenkins.md:167]: @lzdanski how to find these values is already defined in the same page under the [prerequisite section](https://docs.astronomer.io/astro/ci-cd-templates/jenkins?tab=multibranch#prerequisites)

**Comment by jwitz** [comment]
@lzdanski Sorry this is a niche part of the site that I'd hoped you never had to encounter. Jenkins is one of the few pipeline tools left that requires users to `curl` specific versions of the CLI in the pipeline, rather than always using latest. We've had to use sitewide environment variables to ensure that these pipelines are always updated when we release a new version of the Astro CLI.

A negative side effect of this implementation is that any raw code starting with `$` needs to be formatted so that Docusaurus doesn't try to interpret it as a non-existent environment variable. You'll see I made a few additional variables to reformat what was before a raw `$`-led value. @pohek321 Can you check the preview site to make sure that all of the indentation appears as you'd hoped?

File: astro/ci-cd-templates/jenkins.md [hunk 4]
```diff
@@ -145,6 +147,59 @@ To automate code deploys across multiple Deployments using [Jenkins](https://www
...
+            }
+        }
+        post {
+            always {
+                cleanWs()
+            }
+        }
+    }`}</code></pre>
+
+    This `Jenkinsfile` triggers a code push to Astro every time a commit or pull request is merged to the `main` branch of your repository.
```
Comment (lzdanski) [inline astro/ci-cd-templates/jenkins.md:199]:
```suggestion
This `Jenkinsfile` triggers a code push to Astro every time a commit or pull request is merged to the `main` branch of your repository.
```

This is going to sound strange, but did you indent with tabs or spaces? Sometimes Markdown syntax has tab = plain text and 4 spaces = code formatting. I haven't seen it pop up in a while, but it looks like that's happening here. We can either remove the indent or you can try using a tab (which for some reason Github suggestions won't allow me to do).

Comment (pohek321) [reply astro/ci-cd-templates/jenkins.md:199]: Not sure, I'll commit up a change with this commit suggestion though

**Comment by pohek321** [comment]
> @lzdanski Sorry this is a niche part of the site that I'd hoped you never had to encounter. Jenkins is one of the few pipeline tools left that requires users to `curl` specific versions of the CLI in the pipeline, rather than always using latest. We've had to use sitewide environment variables to ensure that these pipelines are always updated when we release a new version of the Astro CLI.
> 
> A negative side effect of this implementation is that any raw code starting with `$` needs to be formatted so that Docusaurus doesn't try to interpret it as a non-existent environment variable. You'll see I made a few additional variables to reformat what was before a raw `$`-led value. @pohek321 Can you check the preview site to make sure that all of the indentation appears as you'd hoped?

@jwitz yeah, it looks good!

**Review by jwitz — APPROVED** [review]

**Pull request merged**